File: 7/largest_num_at_least_twice_than_others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        largest, second_largest, largest_index = -1, -1, -1
        for i in range(len(nums)):
            if nums[i]>largest:
                second_largest = largest
                largest = nums[i]
                largest_index = i
            elif largest>nums[i]>second_largest:
                second_largest=nums[i]
        return largest_index if largest>=2*second_largest else -1

File: 7/largest_num_at_least_twice_than_others.go
func dominantIndex(nums []int) int {
    largest, secondLargest, largestIndex := -1, -1, -1
    for i:=0;i<len(nums);i++{
        if nums[i]>largest {
            secondLargest = largest
            largest = nums[i]
            largestIndex = i
        } else if nums[i] < largest && nums[i] > secondLargest {
            secondLargest = nums[i]
        }
    }
    if largest >= 2*secondLargest {
        return largestIndex
    } else {
        return -1
    }
}
